src/scene.ts

```diff
@@ -4045,7 +4045,10 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
     }
 
     private _processSubCameras(camera: Camera, bindFrameBuffer = true): void {
-        if (camera.cameraRigMode === Constants.RIG_MODE_NONE || (camera.outputRenderTarget && camera.outputRenderTarget.getViewCount() > 1 && this.getEngine().getCaps().multiview)) {
+        if (camera.cameraRigMode === Constants.RIG_MODE_NONE || camera._renderingMultiview()) {
+            if (camera._renderingMultiview() && !this._multiviewSceneUbo) {
+                this._createMultiviewUbo();

```

Comment:
This makes sense, it would also probably improve the `scene.getSceneUniformBuffer` function since right now once you turn multiview on you can't turn it off (from the point of view of that function), unless that doesn't matter and we can always just make the scene uniform buffer contain the two extra matrices for right eye view & projection

Reply:
Perhaps now that the multiview check is less expensive it's ok?

Reply:
do you refer to changing the getSceneUniformBuffer function?

Reply:
I think we can either create the multiview UBO at scene construction if multiview is supported by the engine, or do this check every frame, which is now at most as expensive as the check that already existed, since now it's just a flag on the camera instead of a full function

Reply:
if i remember correctly we already had a multiview ubo implementation:

https://github.com/BabylonJS/Babylon.js/blob/master/src/Engines/Extensions/engine.multiview.ts#L126

Will we be asble to use this implementation here as well? does it make sense?

Reply:
With the WebVR multiview path, we have the exact same behavior, where `_createMultiviewUbo` is called when the camera starts rendering with multiview:
https://github.com/BabylonJS/Babylon.js/blob/87352cefe36b9a6337c344cc1fdfe26859b42b9b/src/Engines/Extensions/engine.multiview.ts#L184-L186
^ `_renderMultiviewToSingleView` runs per-frame as long as `camera._useMultiviewToSingleView` is true.

I'm doing the same thing here as the `_renderMultiviewToSingleView` function, except we don't need to do any of the other stuff since the resizing, outputRenderTarget, and post processing are either taken care of by the composition layer render target provider + webxr camera or they're not necessary (no need to copy the textures to a single one).

If we don't want this check to run per-frame, we could also just have the scene UBO contain the extra matrix that the multiview UBO contains (`viewProjectionR`) and never need to run the check in the first place, or need to explicitly create the multiview UBO.

Reply:
I like the fact that the scene component hasn't been changed. let's go with that without changing the architecture too much.